sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,440 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { HttpOperationResponse } from "@azure/core-http";
+import * as Constants from "../util/constants";
+import {
+  serializeToAtomXmlRequest,
+  deserializeAtomXmlResponse,
+  AtomXmlSerializer
+} from "../util/atomXmlHelper";
+import {
+  getIntegerOrUndefined,
+  getJSONOrUndefined,
+  getStringOrUndefined,
+  getBooleanOrUndefined
+} from "../util/utils";
+import { CorrelationFilter } from "../core/managementClient";
+
+/**
+ * @ignore
+ * Builds the rule options object
+ * @param name
+ * @param ruleOptions
+ */
+export function buildRuleOptions(name: string, ruleOptions: RuleOptions = {}): InternalRuleOptions {
+  const internalRuleOptions: InternalRuleOptions = Object.assign({}, ruleOptions, { name: name });
+  return internalRuleOptions;
+}
+
+/**
+ * @ignore
+ * Builds the rule object
+ * @param rawRule
+ */
+export function buildRule(rawRule: any): Rule | {} {
+  if (rawRule == undefined || rawRule == {}) {
+    return {};
+  } else {
+    const result: Rule = {
+      ruleName: rawRule["RuleName"],
+      topicName: rawRule["TopicName"],
+      subscriptionName: rawRule["SubscriptionName"],
+      filter: getTopicFilterOrUndefined(rawRule["Filter"]),
+      action: getRuleActionOrUndefined(rawRule["Action"]),
+      createdAt: rawRule["CreatedAt"]
+    };
+    return result;
+  }
+}
+
+/**
+ *  @ignore
+ * Helper utility to retrieve `filter` value from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+function getTopicFilterOrUndefined(value: any): SqlFilter | CorrelationFilter | undefined {
+  if (value == undefined) {
+    return undefined;
+  } else {
```

Comment:
nit: You can reduce the indentation by removing the else statement.

```typescript

function getTopicFilterOrUndefined(value: any): SqlFilter | CorrelationFilter | undefined {
	if (value == undefined) {
	    return undefined;
	}

	let result: SqlFilter | CorrelationFilter | undefined;

	if (value["SqlExpression"] != undefined) {
	  result = {
	    sqlExpression: value["SqlExpression"],
	    sqlParameters: getSqlParametersOrUndefined(value["Parameters"]),
	    compatibilityLevel: getIntegerOrUndefined(value["CompatibilityLevel"]),
	    requiresPreprocessing: getBooleanOrUndefined(value["RequiresPreprocessing"])
	};

...
```

Reply:
updated